src/transforms/api/_transform.py

```diff
@@ -356,4 +553,5 @@ def open(self, path, mode="w", **kwargs):  # noqa: A003
             IO:
 
         """
+        self._fs.makedirs(os.path.dirname(path), recreate=True)
```

Comment:
why is this required now?

Reply:
My bad, I should've highlighted it, it's an unrelated fix.

If I take this transform:
```py
@lightweight
@transform(
    my_input=Input("ri.foundry.main.dataset.888a07e8-1d07-427a-9bc4-b4dc1c40708a"),
    my_output=Output('ri.foundry.main.dataset.2375002f-8ee0-4ceb-a182-f573eb80f8f3')
)
def copy_files(my_input, my_output):
    logging.info("hello world")  # check out the build logs on the build report page

    for file in my_input.filesystem().ls():
        with my_input.filesystem().open(file.path, "rb") as f1:
            with my_output.filesystem().open(file.path, "wb") as f2:
                f2.write(f1.read())
```

When running the Preview with fdt, I would get:
```
(base) ➜  Lightweight-examples git:(master) ✗ /Users/aschmelczer/miniconda3/bin/python /Volumes/git/Lightweight-examples/transforms-python/src/myproject/datasets/api.py
/Volumes/git/foundry-dev-tools/src/transforms/api/_decorators.py:80: UserWarning: Setting resources for @lightweight transforms will have no effect when running locally.
  warnings.warn(
Traceback (most recent call last):
  File "/Volumes/git/Lightweight-examples/transforms-python/src/myproject/datasets/api.py", line 44, in <module>
    print(copy_files.compute())
          ^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/git/foundry-dev-tools/src/transforms/api/_transform.py", line 97, in compute
    return handlers[self._type]()
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/git/foundry-dev-tools/src/transforms/api/_transform.py", line 178, in _compute_lightweight
    self(**kwargs)
  File "/Volumes/git/foundry-dev-tools/src/transforms/api/_transform.py", line 81, in __call__
    return self._compute_func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/git/Lightweight-examples/transforms-python/src/myproject/datasets/api.py", line 17, in copy_files
    with my_output.filesystem().open(file.path, "wb") as f2:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/git/foundry-dev-tools/src/transforms/api/_transform.py", line 552, in open
    return self._fs.open(path, mode, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/aschmelczer/miniconda3/lib/python3.11/site-packages/fs/base.py", line 1228, in open
    bin_file = self.openbin(path, mode=bin_mode, buffering=buffering)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/aschmelczer/miniconda3/lib/python3.11/site-packages/fs/memoryfs.py", line 513, in openbin
    raise errors.ResourceNotFound(path)
fs.errors.ResourceNotFound: resource 'spark/part-00002-4e7e654b-c712-4a0f-afdd-d7e89e4425e8-c000.snappy.parquet' not found
```

However, in Foundry, this is allowed and builds successfully (both for Lightweight and regular transforms).

Reply:
Makes sense, can you modify the test `test_transform_one_input`? I think the line `output1.filesystem()._fs.makedir("subfolder")` can now be removed and the test should still stay green.

Reply:
Yes, it does pass now without that line!